docs/IE_PLUGIN_DG/NewTransformation.md

```diff
@@ -1,15 +1,386 @@
+Transformations library is independent from InferenceEngine target library and located in `inference-engine/src/transformations` directory.
+Transformations root directory contains two folders:
+1. ngraph_ops - legacy opset operations needed for nGraph to CNNNetwork conversion.
+2. transformations - includes all transformations, utils, runtime info attributes and pass managers.
 
-bla-bla
+Transformation flow in transformation library has several layers:
+1. Pass managers - executes list of transformations using `*_tbl.hpp` file. For example conversion form OpSetX to OpSetY.
+2. Transformations - performs particular transformation algorithm on `ngraph::Funcion` (find more about transformations in [Transformations types]).
+3. Low level functions that takes set of nodes and performs some transformation action. They are not mandatory and all transformation code can be located inside transformation but if some transformation parts can potentially be reused in other transformations we suggest to keep them as a separate functions.
 
-// ! [new_transformation:part1]
+To decide where to store your transformation code please follow these rules:
+1. If it's plugin specific transformation and can't be reused by other plugins keep source code inside plugin.
+2. If this transformation relates to OpSetXToOpSetY conversion or it's common optimization then keep sources inside transformation library.
 
-And this file must refer to that code via
+After you decided where to store your transformation code you can start develop your own nGraph transformation.
 
-@snippet src/template_transformation.cpp new_transformation:part1
+## Table of Contents:
+
+1. `ngraph::Function` and graph representation
+2. Transformations types
+3. Pattern matching
+4. Working with ngraph::Function
+5. Transformation writing essentials
+6. Common mistakes in transformations
+7. Using pass manager
+8. How to debug transformations
+9. Disabling/Enabling specific transformations for plugin X
+10. Transformations testing
+
+## ngraph::Function and graph representation
+
+nGraph function is a very simple thing: it stores shared pointers to [Result] and [Parameter] operations that are inputs and outputs of the graph. 
+All other operations hold each other via shared pointers: child operation holds its parent (hard link). If operation has no consumers and it's not Result operation
+(shared pointer counter is zero) then it will be destructed and won't be accessible anymore. Each operation in `ngraph::Function` is a shared_ptr and has `ngraph::Node` as a base class.
+
+Below you can find examples how `ngraph::Function` can be created:
+
+@snippet example_ngraph_utils.cpp ngraph_utils:simple_function
+
+@snippet example_ngraph_utils.cpp ngraph_utils:advanced_function
+
+## Transformations types
+
+There are two main transformation types:
+
+###1. ngraph::pass::FunctionalPass
+
+ngraph::pass::FunctionalPass is used for transformations that take entire `ngraph::Function` as input and process it.
```

Comment:
can we enclose all code in this docs into ``? like `ngraph::pass::FunctionalPass` or `std::shared_ptr`

Reply:
I added additional example code into transformation body